src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -0,0 +1,133 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+from __future__ import print_function
+import getpass
+import time
+import subprocess
+import json
+import os
+import sys
+
+from applicationinsights import TelemetryClient
+from applicationinsights.exceptions import enable
+from azure.cli.core import __version__ as core_version
+
+_DEBUG_TELEMETRY = 'AZURE_CLI_DEBUG_TELEMETRY'
+client = {}
+data = []
+APPLICATION = {}
+ARGCOMPLETE_ENV_NAME = None
+
+def set_application(application, argcomplete_env_name):
+    global APPLICATION #pylint: disable=global-statement
```

Comment:
we can't import because application needs telemetry, thus making a circular dependency 

---
In reply to: [78662394](https://github.com/Azure/azure-cli/pull/865#discussion_r78662394) [](ancestors = 78662394)